router/batchrouter/asyncdestinationmanager/sftp/utils.go

```diff
@@ -0,0 +1,323 @@
+package sftp
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/sftp"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+// createSSHConfig creates SSH configuration based on destination
+func createSSHConfig(destination *backendconfig.DestinationT) (*sftp.SSHConfig, error) {
+	destinationConfigJson, err := json.Marshal(destination.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	var config destConfig
+	if err := json.Unmarshal(destinationConfigJson, &config); err != nil {
+		return nil, err
+	}
```

Comment:
Can we add some more context in here.
```suggestion
	destinationConfigJson, err := json.Marshal(destination.Config)
	if err != nil {
		return nil, fmt.Errorf("marshalling destination config: %w", err)
	}

	var config destConfig
	if err := json.Unmarshal(destinationConfigJson, &config); err != nil {
		return nil, fmt.Errorf("unmarshalling destination config: %w", err)
	}
```

Reply:
addressed